tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -167,58 +222,34 @@ spec:
 	})
 
 	Context("When operator backed services are created", func() {
-		It("should list the services if they exist", func() {
-			// First let's grab the etcd operator's name from "odo catalog list services" output
-			operators := helper.CmdShouldPass("odo", "catalog", "list", "services")
-			etcdOperator := regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]`).FindString(operators)
-
-			stdOut := helper.CmdShouldPass("odo", "service", "create", etcdOperator, "--crd", "EtcdCluster", "--dry-run")
-
-			// change the metadata.name from example to example3 so that we can run tests parallely
-			lines := strings.Split(stdOut, "\n")
-			for i, line := range lines {
-				if strings.Contains(line, "name: example") {
-					lines[i] = strings.Replace(lines[i], "example", "example3", 1)
-				}
-			}
-			stdOut = strings.Join(lines, "\n")
 
-			// stdOut contains the yaml specification. Store it to a file
-			randomFileName := helper.RandString(6) + ".yaml"
-			fileName := filepath.Join("/tmp", randomFileName)
-			if err := ioutil.WriteFile(fileName, []byte(stdOut), 0644); err != nil {
-				fmt.Printf("Could not write yaml spec to file %s because of the error %v", fileName, err.Error())
-			}
-
-			// now create operator backed service
-			helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName)
+		JustBeforeEach(func() {
+			preSetup()
+		})
 
-			// now verify if the pods for the operator have started
-			pods := helper.CmdShouldPass("oc", "get", "pods", "-n", CI_OPERATOR_HUB_PROJECT)
-			// Look for pod with example name because that's the name etcd will give to the pods.
-			etcdPod := regexp.MustCompile(`example3-.[a-z0-9]*`).FindString(pods)
+		JustAfterEach(func() {
+			cleanPreSetup()
+		})
 
-			ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", CI_OPERATOR_HUB_PROJECT}
-			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
-				return strings.Contains(output, "Running")
-			})
+		It("should list the services if they exist", func() {
+			createEtcdClusterService()
 
-			stdOut = helper.CmdShouldPass("odo", "service", "list")
+			stdOut := helper.CmdShouldPass("odo", "service", "list")
 			Expect(stdOut).To(ContainSubstring("example"))
 			Expect(stdOut).To(ContainSubstring("EtcdCluster"))
```

Comment:
Make it in a single check as you have done in https://github.com/openshift/odo/pull/3239/files#diff-49f09fb5341717455c0421b3c592e685R243

Reply:
Not touched in this PR. Please open a separate issue for this to be addressed. I can't have double standards between this and https://github.com/openshift/odo/pull/3239#discussion_r429256479 even if I'm aware of the change that needs to be made. I'm sorry.

Reply:
@dharmit Do you need a separate issue for this?

Reply:
Issue https://github.com/openshift/odo/issues/3245